tests/phpunit/tests/test-olt-manager.php

```diff
@@ -21,4 +30,44 @@ public function test_polylang_first() {
 		delete_option( 'active_plugins' );
 		delete_option( 'active_sitewide_plugins' );
 	}
+
+	public function test_load_textdomains() {
+		// Clean up!
+		unset( $GLOBALS['wp_themes'], $GLOBALS['l10n'], $GLOBALS['l10n_unloaded'] );
+
+		/** @var WP_Textdomain_Registry $wp_textdomain_registry */
+		global $wp_textdomain_registry;
+
+		$wp_textdomain_registry = new WP_Textdomain_Registry();
+
+		update_option( 'WPLANG', '' );
+
+		// Copy language file.
+		@mkdir( DIR_TESTDATA );
+		@mkdir( WP_LANG_DIR );
+		@mkdir( WP_LANG_DIR . '/plugins' );
+		copy( PLL_TEST_DATA_DIR . 'fr_FR.mo', WP_LANG_DIR . '/plugins/foo-fr_FR.mo' );
+
+		new PLL_OLT_Manager();
+		$options                 = PLL_Install::get_default_options();
+		$options['force_lang']   = 0;
+		$options['default_lang'] = 'en';
+		$model                   = new PLL_Model( $options );
+		$links_model             = $model->get_links_model();
+		$frontend                = new PLL_Frontend( $links_model );
+
+		$is_loaded = load_plugin_textdomain( 'foo' );
+
+		$this->assertFalse( $is_loaded, 'textdomain should be bypassed by PLL_OLT_Manager.' );
```

Comment:
Using `load_plugin_texdomain()` disallows to treproduce the bug. We should always use `_load_textdomain_just_in_time()` even before the language is defined. As far as I remember this the removal of `load_theme_texdomain()` in Twenty Twenty one which revealed the bug:
 ```suggestion
		$this->assertSame( 'Dashboard', __( 'Dashboard', 'foo' ) );
```

Reply:
Indeed, I managed to reproduce the issue with the old version of `PLL_OLT_Manager` and a call to `__()` before the language is defined!
See 0774f4e6
Even though it's weird that `load_theme_textdomain()` is still called in TT1 from the beginning ([cf](https://github.com/WordPress/twentytwentyone/blame/ba9f20cad89163761185c0467b346ba42541ae22/functions.php#L36)

Reply:
It used to be the case at the beginning. But [not anymore](https://github.com/WordPress/wordpress-develop/commit/1d1a2e3dec64bca6b94abd0eaa871f3916bbc11b).

Reply:
What?! They commit directly in `wordpress-develop` instead of the dedicated repo?
I learned something, thanks!